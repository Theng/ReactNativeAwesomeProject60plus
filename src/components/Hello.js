import React, { useState, useCallback, useMemo } from "react";
import { View, Text } from "react-native";
import INC from "./INC";
function MyComponent({ label }) {
    console.log("Render Hello Compoenent.");

    const [number, increase] = useState(0);
    const [a, setAA] = useState(1);
    const [b, setB] = useState(1);

    const increment = useCallback(
        x => {
            increase(n => n + x);
        },
        [increase]
    );

    const incrementA = useCallback(
        x => {
            setAA(n => n + x);
        },
        [setAA]
    );
    const incrementB = useCallback(
        x => {
            setB(n => n + x);
        },
        [setB]
    );

    const mySum = useMemo(() => sum(a, b), [a, b]);

    return (
        <View>
            <Text>useCallback: {label + ": " + number}</Text>
            <INC label="Increase Hello" increment={increment} />
            <Text>useMemo: sum (a+b): {mySum}</Text>
            <INC label="Increase A" increment={incrementA} />
            <INC label="Increase B" increment={incrementB} />
        </View>
    );
}

export default React.memo(MyComponent);

const sum = (a, b) => {
    console.log("computing SUM:::");
    return a + b;
};
