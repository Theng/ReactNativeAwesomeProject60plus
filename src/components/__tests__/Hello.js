import React from 'react';
import Hello from '../Hello';
import {render, cleanup} from 'react-native-testing-library';

afterEach(cleanup);

describe('<Hello />', () => {
  it('should match snapshot', () => {
    const rendered = render(<Hello />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should show label abacaba', () => {
    const rendered = render(<Hello label={'abacaba'} />);
    const textComponent = rendered.getByTestId('text');
    expect(textComponent.props.children).toEqual('abacaba');
  });
});