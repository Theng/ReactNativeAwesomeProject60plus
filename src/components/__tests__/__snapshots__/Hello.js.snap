// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should match snapshot 1`] = `
<View>
  <Text
    testID="text"
  />
  <Text>
    useCallback: 
    undefined: 0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increase Hello
      </Text>
    </View>
  </View>
  <Text>
    useMemo: sum (a+b): 
    2
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increase A
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increase B
      </Text>
    </View>
  </View>
</View>
`;
